pkg/resource/alias/hooks.go

```diff
@@ -80,20 +80,29 @@ func (rm *resourceManager) updateProvisionedConcurrency(
 	exit := rlog.Trace("rm.updateProvisionedConcurrency")
 	defer exit(err)
 
+	// Check if the user deleted the 'ProvisionedConcurrency' configuration
+	// If yes, delete ProvisionedConcurrencyConfig
+	if desired.ko.Spec.ProvisionedConcurrencyConfig == nil {
+		input_delete := &svcsdk.DeleteProvisionedConcurrencyConfigInput{
+			FunctionName: aws.String(*desired.ko.Spec.FunctionName),
+			Qualifier:    aws.String(*desired.ko.Spec.Name),
+		}
+		_, err = rm.sdkapi.DeleteProvisionedConcurrencyConfigWithContext(ctx, input_delete)
+		rm.metrics.RecordAPICall("DELETE", "DeleteProvisionedConcurrency", err)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
 	dspec := desired.ko.Spec
```

Comment:
It's not a big deal, but considering that we're defining this variable, we should move it to a little earlier, so we can use `dspec` in the previous `if` as well.

Reply:
It was defined in the beginning only, but after discussion with Amine, him and I decided to keep the delete part in the beginning before we even define any of the variables used for the `Update` part of the field. The purpose was to reduce the doubt, as the function is for `UpdateProvisionedConcurrency` and we are handling the deletion too for the case when there isn't any `ProvisionedConcurrency`.